### PR TITLE
ARROW-10059: [R][Doc] Give more advice on how to set up C++ build

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

File: r/README.md [hunk 1]
```diff
@@ -102,6 +102,43 @@ elsewhere, you’ll need to build it from source too.
 
 First, install the C++ library. See the [developer
 guide](https://arrow.apache.org/docs/developers/cpp/building.html) for details.
+It's recommended to make a `build` directory inside of the `cpp` directory of
+the Arrow git repository (it is git-ignored). Assuming you are inside `cpp/build`,
+you'll first call `cmake` to configure the build and then `make install`.
```
Comment (pitrou) [inline r/README.md:107]: Are you advocatin "make"? Even on Windows?

File: r/README.md [hunk 2]
```diff
@@ -102,6 +102,43 @@ elsewhere, you’ll need to build it from source too.
...
+  -DARROW_WITH_BZ2=ON \
+  -DARROW_WITH_LZ4=ON \
+  -DARROW_WITH_SNAPPY=ON \
+  -DARROW_WITH_ZLIB=ON \
+  -DARROW_WITH_ZSTD=ON \
+```
+
+Other flags that may be useful:
+
+* `-DARROW_EXTRA_ERROR_CONTEXT=ON` makes errors coming from the C++ library point to files and line numbers
```
Comment (pitrou) [inline r/README.md:139]: I don't think this is useful except for developers.

File: r/README.md (hunk 1 above)
Comment (xhochy) [reply r/README.md:107]: R development is mostly happening in a MinGW / msys2-based environment on Windows, so `make` would be a default choice.

File: r/README.md (hunk 2 above)
Comment (nealrichardson) [reply r/README.md:139]: This is in the instructions for developers. Regular users shouldn't ever have to mess with cmake flags

Comment (pitrou) [reply r/README.md:139]: Ah, I see. Mea culpa :-)

File: r/README.md (hunk 1 above)
Comment (nealrichardson) [reply r/README.md:107]: I guess I could change it to `cmake --build . --target install`? I'm kinda assuming that if you know how to change the cmake generator then you also know how to invoke it.

Comment (pitrou) [reply r/README.md:107]: >  I'm kinda assuming that if you know how to change the cmake generator then you also know how to invoke it.

Sounds fair.

**Pull request closed without merging**